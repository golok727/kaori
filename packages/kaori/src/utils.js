export function invariant(condition, message) {
    if (!condition) {
        throw new Error(message);
    }
}
//# sourceMappingURL=utils.js.map