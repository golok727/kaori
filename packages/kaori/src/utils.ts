export function invariant(
  condition: boolean,
  message: string
): asserts condition {
  if (!condition) {
    throw new Error(message);
  }
}

export const has_own = Object.hasOwn;
export const get_own_property_descriptor = Object.getOwnPropertyDescriptor;
export const define_property = Object.defineProperty;
