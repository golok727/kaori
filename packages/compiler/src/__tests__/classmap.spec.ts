import { describe, expect, it } from 'vitest';
import { compile } from './utils';

describe('styleMap transformation', () => {
  it('should transform classMap to class attribute with classMap function wrapped', async () => {
    const input = `
function ClassMap() {
  return <div classMap={{ active: true, disabled: false }}>Hello</div>;
}`;

    const output = await compile(input);
    expect(output).toMatchSnapshot();
  });
});
