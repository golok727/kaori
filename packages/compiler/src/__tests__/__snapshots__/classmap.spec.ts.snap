// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`styleMap transformation > should transform classMap to class attribute with classMap function wrapped 1`] = `
"import { html, classMap } from "kaori.js";
function ClassMap() {
  return html\`<div class=\${classMap({
    active: true,
    disabled: false
  })}>Hello</div>\`;
}"
`;
